runhouse/resources/hardware/cluster.py

```diff
@@ -702,61 +700,33 @@ def _sync_runhouse_to_cluster(
         if not self.ips:
             raise ValueError(f"No IPs set for cluster <{self.name}>. Is it up?")
 
-        conda_env_name = self.image.conda_env_name if self.image else None
         self._run_setup_step(
             step=ImageSetupStep(
                 step_type=ImageSetupStepType.PIP_INSTALL,
                 reqs=["ray"],
-                conda_env_name=conda_env_name,
+                conda_env_name=self.conda_env_name,
             ),
             parallel=parallel,
         )
 
         # If local_rh_package_path is provided, install the package from the local path
         if local_rh_package_path:
-            local_rh_package_path = local_rh_package_path.parent
-            dest_path = f"~/{local_rh_package_path.name}"
-
             self._run_setup_step(
                 step=ImageSetupStep(
-                    step_type=ImageSetupStepType.RSYNC,
-                    source=str(local_rh_package_path),
-                    dest=dest_path,
-                    contents=True,
-                    filter_options="- docs/",
+                    step_type=ImageSetupStepType.SYNC_PACKAGE,
+                    package=str(local_rh_package_path),
```

Comment:
Could just be "runhouse" instead of the path and it will search for editable or not automatically, no?